src/util/pointer_offset_size.cpp

```diff
@@ -529,15 +532,16 @@ compute_pointer_offset(const exprt &expr, const namespacet &ns)
   else if(expr.id()==ID_index)
   {
     const index_exprt &index_expr=to_index_expr(expr);
-    const typet &array_type=ns.follow(index_expr.array().type());
     DATA_INVARIANT(
-      array_type.id()==ID_array,
-      "index into array expected, found "+array_type.id_string());
+      index_expr.array().type().id() == ID_array,
```

Comment:
? Are you sure about removing the `ns.follow`, couldn't the array have symbol type here?

Reply:
No, we don't use symbol types for arrays (anymore - I think many many years ago we might).